### PR TITLE
Fix request processor endless loop

### LINKED ISSUE #89: Request Processor endless loop.
### RaftKeeper version
v2.0.1

### Description

```
Has 18 error requests
2023.10.09 21:30:50.817469 [ 44 ] <Warning> RequestProcessor: Try find error request session 0xc0a, xid 3727471, error code -32768
2023.10.09 21:30:50.817500 [ 44 ] <Warning> RequestProcessor: Session 0xc0a, no pending requests
2023.10.09 21:30:50.817529 [ 44 ] <Warning> RequestProcessor: Not found error request session 0xc0a, xid 3727471 from pending queue. Maybe it is still in the request queue and will be processed next time
2023.10.09 21:30:50.817660 [ 44 ] <Warning> RequestProcessor: Has 18 error requests
2023.10.09 21:30:50.817690 [ 44 ] <Warning> RequestProcessor: Try find error request session 0xc0a, xid 3727471, error code -32768
2023.10.09 21:30:50.817722 [ 44 ] <Warning> RequestProcessor: Session 0xc0a, no pending requests
2023.10.09 21:30:50.817752 [ 44 ] <Warning> RequestProcessor: Not found error request session 0xc0a, xid 3727471 from pending queue. Maybe it is still in the request queue and will be processed next time
2023.10.09 21:30:50.818028 [ 44 ] <Warning> RequestProcessor: Has 18 error requests
...
```

This error will cause high CPU usage.

### PR DESCRIPTION
**Pull request by JackyWoo**
### Which issues of this PR fixes:

This PR try to fix #89 

### Change log:

1. Add `KeeperCommon.h` which contains common prototypes such as `RequestForSession`
2. Rename `KeeperCommon.h` to `KeeperUtils.h`
3. Add `RequestId` instead of UInt28(session_id, xid)
4. Fix request processor endless loop
5. `sessionCleanerTask` will not `finishSession`
6. Some code refactoring.

### CODE REVIEW

**Comment by JackyWoo** [comment]
The CI error is data race and will be fixed here #90

File: src/Service/RequestProcessor.cpp [hunk 1]
```diff
@@ -58,8 +61,11 @@ void RequestProcessor::run()
             }
             request_thread->wait();
 
-            /// 3. process committed request, single thread
+            /// 2. process committed request, single thread
             processCommittedRequest(committed_request_size);
+
+            /// 3. process error requests
+            processErrorRequest();
```
Comment (nicelulu) [inline src/Service/RequestProcessor.cpp:68]: Need to get the number of current errors in advance, before moving to the pending queue, so as to ensure that the processed errors have been added to the pending queue.

Comment (JackyWoo) [reply src/Service/RequestProcessor.cpp:68]: Now errors is not a queue, we can't handle n items in order. Maybe we should execute errors in order.

File: src/Service/RequestProcessor.cpp [hunk 2]
```diff
@@ -92,280 +97,254 @@ void RequestProcessor::moveRequestToPendingQueue(RunnerId runner_id)
...
+        {
+            {
+                std::unique_lock lk(mutex);
+                found_error = errors.contains(first_pending_request.getRequestId());
+            }
+
+            if (found_error)
+                LOG_WARNING(log, "Found error request, We should terminate the processing of committed(write) requests.");
+            else
+                process_not_in_pending_queue();
```
[the remark this answers is not in the document]
Comment (JackyWoo) [reply src/Service/RequestProcessor.cpp:161]: Here `first_pending_request.request->xid != committed_request.request->xid`, maybe there is no need to traverse pending queue.

**Pull request merged**